lib/Service/ParticipantService.php

```diff
@@ -161,12 +166,32 @@ public function joinRoom(Room $room, IUser $user, string $password, bool $passed
 				throw new InvalidPasswordException('Provided password is invalid');
 			}
 
-			// User joining a public room, without being invited
-			$this->addUsers($room, [[
-				'actorType' => Attendee::ACTOR_USERS,
-				'actorId' => $user->getUID(),
-				'participantType' => Participant::USER_SELF_JOINED,
-			]]);
+			// User joining a group call through listing
+			if (
+				$room->getType() === Room::GROUP_CALL && (
+					// this check should have happened earlier already but let's stay defensive
+					$room->getListable() === Room::LISTABLE_ALL || (
+						$room->getListable() === Room::LISTABLE_USERS && !$this->groupManager->isInGroup($user->getUID(), 'guest_app')
```

Comment:
use the function from the OCA\Talk\Manager instead, so we don't have multiple places with this magic string?

Reply:
I couldn't inject Manager without causing an infinite loop of dependencies.

is there an approach for this ?

Reply:
Add the function to another service which can be injected in both, but well too much effort. just keep it like this for now.